e2e/common_test.go

```diff
@@ -34,3 +34,28 @@ func ensureRemediationResourceExists(name string, namespace string, remediationR
 		return fmt.Errorf("not found")
 	}
 }
+
+func ensureRemediationResourceDoesNotExist(nodeName string, namespace string, remediationResource schema.GroupVersionResource) func() error {
+	return func() error {
+		// The CR name doesn't always match the node name in case multiple CRs of same type for the same node are supported
+		// So list all, and look for the node name in the annotation
+		list, err := dynamicClient.Resource(remediationResource).Namespace(namespace).List(context.Background(), metav1.ListOptions{})
+		if err != nil {
+			if errors.IsNotFound(err) {
+				log.Info("verified remediation resource does not exist")
+				return nil
+			} else {
+				log.Error(err, "failed to get remediation resource")
+			}
+			return err
+		}
+		for _, cr := range list.Items {
+			if annotationNodeName, exists := cr.GetAnnotations()[commonannotations.NodeNameAnnotation]; exists && annotationNodeName == nodeName {
+				log.Info("remediation resource still exist")
+				return fmt.Errorf("remediation exist")
```

Comment:
I know it's just a test, but I think that enhancing the error message with info about which remediation was found might be helpful.